plugins/destination/duckdb/client/read.go

```diff
@@ -98,59 +97,47 @@ func reverseTransformRecord(sc *arrow.Schema, rec arrow.Record) arrow.Record {
 }
 
 func reverseTransformArray(dt arrow.DataType, col arrow.Array) arrow.Array {
-	switch {
-	case arrow.TypeEqual(dt, types.ExtensionTypes.UUID):
-		return reverseTransformUUID(col.(*array.FixedSizeBinary))
-	case arrow.TypeEqual(dt, types.ExtensionTypes.Inet):
-		return reverseTransformInet(col.(*array.String))
-	case arrow.TypeEqual(dt, types.ExtensionTypes.MAC):
-		return reverseTransformMAC(col.(*array.String))
-	case arrow.TypeEqual(dt, arrow.PrimitiveTypes.Uint16):
+	if plainWrite(dt) {
+		return col
+	}
+
+	switch dt := dt.(type) {
+	case *types.UUIDType:
+		return array.NewExtensionArrayWithStorage(dt, col.(*array.FixedSizeBinary))
```

Comment:
duckdb reads it as fixed size binary (and we can just use the storage as that's the same in our UUID implementation)